### ISSUE TITLE
[Migration Vue 3] The module does not seem to initialize

### ISSUE THREAD

**Issue by antt0n** [stephaneLeroy/vue-erdjs#1]
Hello,
I have a problem with the module.
Below is the console error I get in my browser:
`Uncaught ReferenceError: global is not defined
    at node_modules/@elrondnetwork/erdjs/out/globals.js (globals.ts:1:1)
    at __require2 (chunk-OGZJHWXM.js?v=1d93d117:19:50)
    at node_modules/@elrondnetwork/erdjs/out/index.js (index.ts:1:1)
    at __require2 (chunk-OGZJHWXM.js?v=1d93d117:19:50)
    at index.min.js:1:83
    at node_modules/vue-erdjs/dist/index.min.js (index.min.js:1:339)
    at __require2 (chunk-OGZJHWXM.js?v=1d93d117:19:50)
    at dep:vue-erdjs:1:16`

Chrome 64 / Win 11 64
Vue 3.2.29
erdjs 9.0.3

**Comment by antt0n** [comment]
Ce problème semble évident, puisque le projet n'est pas encore compatible avec Vue3. 
Y a-t-il une date provisoire ?
Une autre branche privée ?

**Comment by antt0n** [comment]
Hey !
Any news about the compatibility with Vue 3 ?
Thanks

**Comment by Schahin-R** [comment]
Hello, we are experiencing the same error on:
Vue 3.2.25
erdjs 9.2.4

We would like to help to resolve that issue, so if you can guide us, we would like to help.

**Comment by stephaneLeroy** [comment]
Hello. Thank you for your feedback.
Work in progress (vue3-migration branch)

**Comment by RoobiPro** [comment]
Hello @stephaneLeroy
How is the vue3 migration going? Any estimation when it's going to be ready?
Thank you

**Comment by stephaneLeroy** [comment]
You can follow the migration on "vue3-migration" branch.
The Vite build part is boilerplate as erdjs include Node module we need to polyfill and I have still issue with it

**Comment by NiftyWell** [comment]
Hello, I have a problem when launching the vue-erdjs-demo project with vue-erdjs@1.0.0-vue3-erdjs11-rc2

```
Forced re-optimization of dependencies
error when starting dev server:
Error: Failed to resolve entry for package "vue-erdjs". The package may have incorrect main/module/exports specified in its package.json.
    at packageEntryFailure (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:34380:11)
    at resolvePackageEntry (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:34377:5)
    at tryNodeResolve (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:34135:20)
    at Context.resolveId (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:33937:28)
    at Object.resolveId (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:35184:55)
    at async file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:62424:21
    at async addManuallyIncludedOptimizeDeps (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:36994:31)
    at async createDepsOptimizer (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:36195:9)
    at async initDepsOptimizer (file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:36117:9)
    at async file:///home/my_user/Documents/Vue-3/vue-erdjs-vue3-migration/vue-erdjs-demo/node_modules/vite/dist/node/chunks/dep-561c5231.js:59135:17
npm ERR! Lifecycle script `dev` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: vue-erdjs-demo@0.0.0 
npm ERR!   at location: /home/well/Documents/DEV/TEST/vue-erdjs-vue3-migration/vue-erdjs-demo 
```
main/module/exports of vue-erdjs:
```json
"main": "./dist/index.umd.js",
"module": "./dist/index.es.js",
"exports": {
    ".": {
        "import": "./dist/index.es.js",
        "require": "./dist/index.umd.js"
    },
    "./dist/index.css": "./dist/index.css",
    "./vite/vite.config.node.js": "./vite/vite.config.node.js"
},
```
Ubuntu 20.04.4 LTS
vue-erdjs@1.0.0-vue3-erdjs11-rc2

**Comment by stephaneLeroy** [comment]
Vue 3 migration is now available on npm (version 1.0.0-vue3-erdjs11-rc2)

Let me know if you have issue with it. 
You can check the demo to see how to configure vite build. This is necessary because erdjs include packages that depend on nodejs core libraries (not available in browser / vite)
